Debian #556797
LP #454487
GNOME #600188

--- libgnome-desktop/gnome-rr.c.orig	2009-11-18 12:39:32.274397008 +0100
+++ libgnome-desktop/gnome-rr.c	2009-11-18 12:53:22.690898043 +0100
@@ -630,6 +630,13 @@
 
 	XRRQueryVersion (dpy, &screen->rr_major_version, &screen->rr_minor_version);
 
+	if (screen->rr_major_version != 1 || screen->rr_minor_version < 2) {
+	    g_set_error (error, GNOME_RR_ERROR, GNOME_RR_ERROR_NO_RANDR_EXTENSION,
+		         _("RANDR extension version 1.2 or later is not present"));
+	    g_free (screen);
+	    return NULL;
+	}
+
 	screen->info = screen_info_new (screen, TRUE, error);
 	
 	if (!screen->info) {
