diff -Nur -x '*.orig' -x '*~' gnome-desktop-2.27.91/libgnome-desktop/gnome-desktop-item.c gnome-desktop-2.27.91.new/libgnome-desktop/gnome-desktop-item.c
--- gnome-desktop-2.27.91/libgnome-desktop/gnome-desktop-item.c	2009-04-20 03:39:06.000000000 +1000
+++ gnome-desktop-2.27.91.new/libgnome-desktop/gnome-desktop-item.c	2009-08-27 10:47:12.000000000 +1000
@@ -83,6 +83,7 @@
 	GHashTable *main_hash;
 
 	char *location;
+    char *gettext_domain;
 
 	time_t mtime;
 
@@ -383,6 +384,7 @@
 				       "1.0");
 
 	retval->launch_time = 0;
+    retval->gettext_domain = NULL;
 
 	return retval;
 }
@@ -544,6 +546,12 @@
 	return retval;
 }
 
+/*
+ * ZK: I needed lookup before the real declaration so here's a prototype
+ */
+static const char *
+lookup (const GnomeDesktopItem *item, const char *key);
+
 /**
  * gnome_desktop_item_new_from_uri:
  * @uri: URI to load the GnomeDesktopItem from
@@ -679,6 +687,7 @@
 
 	g_object_unref (subfn);
 
+    retval->gettext_domain = lookup(retval, GNOME_DESKTOP_ITEM_GETTEXT_DOMAIN);
 	return retval;
 }
 
@@ -722,6 +731,7 @@
 	
 	/* FIXME: Sort order? */
 
+    retval->gettext_domain = lookup(retval, GNOME_DESKTOP_ITEM_GETTEXT_DOMAIN);
 	return retval;
 }
 
@@ -991,6 +1001,14 @@
 	    strcmp (locale, "C") == 0) {
 		return lookup (item, key);
 	} else {
+        if (item->gettext_domain) {
+            char *value = lookup(item, key);
+            if (value != NULL && value[0] != '\0') {
+                char *locale_value = dgettext(item->gettext_domain, value);
+                if (locale_value)
+                    return locale_value;
+            }
+        }
 		const char *ret;
 		char *full = g_strdup_printf ("%s[%s]", key, locale);
 		ret = lookup (item, full);
diff -Nur -x '*.orig' -x '*~' gnome-desktop-2.27.91/libgnome-desktop/libgnome/gnome-desktop-item.h gnome-desktop-2.27.91.new/libgnome-desktop/libgnome/gnome-desktop-item.h
--- gnome-desktop-2.27.91/libgnome-desktop/libgnome/gnome-desktop-item.h	2009-04-20 03:39:06.000000000 +1000
+++ gnome-desktop-2.27.91.new/libgnome-desktop/libgnome/gnome-desktop-item.h	2009-08-27 10:47:12.000000000 +1000
@@ -95,6 +95,7 @@
 #define GNOME_DESKTOP_ITEM_SORT_ORDER	"SortOrder" /* strings */
 #define GNOME_DESKTOP_ITEM_URL		"URL" /* string */
 #define GNOME_DESKTOP_ITEM_DOC_PATH	"X-GNOME-DocPath" /* string */
+#define GNOME_DESKTOP_ITEM_GETTEXT_DOMAIN	"X-Ubuntu-Gettext-Domain" /* string */
 
 /* The vfolder proposal */
 #define GNOME_DESKTOP_ITEM_CATEGORIES	"Categories" /* string */
