Description: Gettext support for desktop files
Author: ?

Index: gnome-desktop-2.32.0/libgnome-desktop/gnome-desktop-item.c
===================================================================
--- gnome-desktop-2.32.0.orig/libgnome-desktop/gnome-desktop-item.c	2010-09-24 11:24:03.000000000 +0200
+++ gnome-desktop-2.32.0/libgnome-desktop/gnome-desktop-item.c	2010-10-15 16:38:41.156592194 +0200
@@ -83,6 +83,7 @@
 	GHashTable *main_hash;
 
 	char *location;
+    char *gettext_domain;
 
 	time_t mtime;
 
@@ -386,6 +387,7 @@
 				       "1.0");
 
 	retval->launch_time = 0;
+    retval->gettext_domain = NULL;
 
 	return retval;
 }
@@ -547,6 +549,12 @@
 	return retval;
 }
 
+/*
+ * ZK: I needed lookup before the real declaration so here's a prototype
+ */
+static const char *
+lookup (const GnomeDesktopItem *item, const char *key);
+
 /**
  * gnome_desktop_item_new_from_uri:
  * @uri: URI to load the GnomeDesktopItem from
@@ -682,6 +690,7 @@
 
 	g_object_unref (subfn);
 
+    retval->gettext_domain = lookup(retval, GNOME_DESKTOP_ITEM_GETTEXT_DOMAIN);
 	return retval;
 }
 
@@ -725,6 +734,7 @@
 	
 	/* FIXME: Sort order? */
 
+    retval->gettext_domain = lookup(retval, GNOME_DESKTOP_ITEM_GETTEXT_DOMAIN);
 	return retval;
 }
 
@@ -994,6 +1004,14 @@
 	    strcmp (locale, "C") == 0) {
 		return lookup (item, key);
 	} else {
+        if (item->gettext_domain) {
+            char *value = lookup(item, key);
+            if (value != NULL && value[0] != '\0') {
+                char *locale_value = dgettext(item->gettext_domain, value);
+                if (locale_value)
+                    return locale_value;
+            }
+        }
 		const char *ret;
 		char *full = g_strdup_printf ("%s[%s]", key, locale);
 		ret = lookup (item, full);
Index: gnome-desktop-2.32.0/libgnome-desktop/libgnome/gnome-desktop-item.h
===================================================================
--- gnome-desktop-2.32.0.orig/libgnome-desktop/libgnome/gnome-desktop-item.h	2010-09-24 11:24:03.000000000 +0200
+++ gnome-desktop-2.32.0/libgnome-desktop/libgnome/gnome-desktop-item.h	2010-10-15 16:38:41.160592196 +0200
@@ -95,6 +95,7 @@
 #define GNOME_DESKTOP_ITEM_SORT_ORDER	"SortOrder" /* strings */
 #define GNOME_DESKTOP_ITEM_URL		"URL" /* string */
 #define GNOME_DESKTOP_ITEM_DOC_PATH	"X-GNOME-DocPath" /* string */
+#define GNOME_DESKTOP_ITEM_GETTEXT_DOMAIN	"X-Ubuntu-Gettext-Domain" /* string */
 
 /* The vfolder proposal */
 #define GNOME_DESKTOP_ITEM_CATEGORIES	"Categories" /* string */
