Description: Handle failure to create/obtain the background pixmap
Bug-Ubuntu: http://launchpad.net/bugs/661283

Index: gnome-desktop-2.32.1/libgnome-desktop/gnome-bg.c
===================================================================
--- gnome-desktop-2.32.1.orig/libgnome-desktop/gnome-bg.c	2010-12-23 17:15:31.563108002 +1100
+++ gnome-desktop-2.32.1/libgnome-desktop/gnome-bg.c	2010-12-23 17:16:39.653108002 +1100
@@ -1115,6 +1115,8 @@
 	else {
 		pixmap = gdk_pixmap_new (window, pm_width, pm_height, -1);
 	}
+	if (pixmap == NULL)
+		return NULL;
 	
 	cr = gdk_cairo_create (pixmap);
 	if (!bg->filename && bg->color_type == GNOME_BG_COLOR_SOLID) {
