diff -Nur -x '*.orig' -x '*~' gnome-desktop-2.28.1/libgnome-desktop/gnome-rr.c gnome-desktop-2.28.1.new/libgnome-desktop/gnome-rr.c
--- gnome-desktop-2.28.1/libgnome-desktop/gnome-rr.c	2009-08-19 16:51:03.000000000 +0100
+++ gnome-desktop-2.28.1.new/libgnome-desktop/gnome-rr.c	2009-10-31 01:11:51.382756270 +0000
@@ -375,6 +375,8 @@
 
     /* First update the screen resources */
 
+	gdk_error_trap_push ();
+
     if (needs_reprobe)
         resources = XRRGetScreenResources (xdisplay, xroot);
     else
@@ -394,6 +396,13 @@
 #endif
     }
 
+	gdk_flush ();
+	if (gdk_error_trap_pop ()) {
+		g_set_error (error, GNOME_RR_ERROR, GNOME_RR_ERROR_UNKNOWN,
+			 	_("unhandled X error while getting the screen resources"));
+	    return FALSE;
+	}
+
     if (resources)
     {
 	if (!fill_screen_info_from_resources (info, resources, error))
