From: Jeremy Bicha <jbicha@ubuntu.com>
Date: Thu, 24 Aug 2017 15:07:01 -0400
Subject: Disable bubblewrap since it's not in main yet

Bug: https://launchpad.net/bugs/1709164
---
 configure.ac | 1 -
 1 file changed, 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 94ade7f..f6d21f5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -177,7 +177,6 @@ case $host_os in
         ;;
     esac
     AC_DEFINE_UNQUOTED(_GNU_SOURCE, 1, [Define to include GNU extensions])
-    AC_DEFINE_UNQUOTED(HAVE_BWRAP, 1, [Define to 1 if Bubblewrap support is available])
     AC_DEFINE_UNQUOTED(INSTALL_PREFIX, "$prefix", [Path to library install prefix])
     ;;
 esac
